File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        
        Map<Integer, Integer> lookup = new HashMap<>();
        
        int max = Integer.MIN_VALUE;
        int ans = nums[0];
        
        for(int i = 0; i < nums.length; i++){
            lookup.put(nums[i], lookup.getOrDefault(nums[i], 0) + 1 );
            if (lookup.get(nums[i]) > max)
            {
                max = lookup.get(nums[i]);
                ans = nums[i];
            }
        }
        
        return ans;
    }
}

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        mapdict = {}
        s = len(nums)
        if s == 1:
            return nums[0]
        for n in nums:
            if n in mapdict.keys():
                mapdict[n] = mapdict.get(n) + 1
                if mapdict.get(n) > s/2:
                    return n
            else:
                mapdict[n] = 1